src/emissions-workspace/emissions.service.ts

```diff
@@ -164,174 +165,183 @@ export class EmissionsWorkspaceService {
-        identifiers,
-        currentTime,
-      ),
-      this.importLongTermFuelFlow(
-        params,
-        monitoringLocations,
-        reportingPeriodId,
-        identifiers,
-        currentTime,
-      ),
-      this.importDailyBackstop(
-        params,
-        monitoringLocations,
-        reportingPeriodId,
-        identifiers,
-        currentTime,
-      ),
-    ];
-
-    const importResults = await Promise.allSettled(importPromises);
-
-    for (const importResult of importResults) {
-      if (importResult.status === 'rejected') {
-        throw new EaseyException(
-          new Error(importResult.reason.toString()),
-          HttpStatus.INTERNAL_SERVER_ERROR,
+        await this.importSummaryValue(
+          params,
+          monitoringLocations,
+          reportingPeriodId,
+          identifiers,
+          currentTime,
+          trx,
+        );
+        await this.importSorbentTrap(
+          params,
+          reportingPeriodId,
+          monitoringLocations,
+          identifiers,
+          currentTime,
+          trx,
+        );
+        await this.importNsps4tSummaries(
+          params,
+          monitoringLocations,
+          reportingPeriodId,
+          identifiers,
+          currentTime,
+          trx,
+        );
+        await this.importWeeklyTestSummary(
+          params,
+          monitoringLocations,
+          reportingPeriodId,
+          identifiers,
+          currentTime,
+          trx,
+        );
+        await this.importLongTermFuelFlow(
+          params,
+          monitoringLocations,
+          reportingPeriodId,
+          identifiers,
+          currentTime,
+          trx,
+        );
+        await this.importDailyBackstop(
+          params,
+          monitoringLocations,
+          reportingPeriodId,
+          identifiers,
+          currentTime,
+          trx,
         );
-      }
-    }
 
-    try {
-      await this.repository.save(
-        this.repository.create({
+        const repository = withTransaction(this.repository, trx);
+        await repository.save(
+          repository.create({
+            monitorPlanId,
+            reportingPeriodId,
+            evalStatusCd: 'EVAL',
+            needsEvalFlag: 'Y',
+            submissionAvailabilityCd: 'GRANTED',
+            lastUpdated: new Date(),
+          }),
+        );
+
+        await repository.updateAllViews(
           monitorPlanId,
-          reportingPeriodId,
-          evalStatusCd: 'EVAL',
-          needsEvalFlag: 'Y',
-          submissionAvailabilityCd: 'GRANTED',
-          lastUpdated: new Date(),
-        }),
-      );
+          params.quarter,
+          params.year,
+        );
 
-      await this.repository.updateAllViews(
-        monitorPlanId,
-        params.quarter,
-        params.year,
-      );
+        //Finally, perform the updates (reset needs eval flag, etc) for those records
+        this.updateCollaterallyAffectedRecords(monitorPlanId, reportingPeriodId);
```

Comment:
This should be updated to use the transaction:
```suggestion
        this.updateCollaterallyAffectedRecords(monitorPlanId, reportingPeriodId, trx);
```

And the method should be updated to take that parameters:
```typescript
  async updateCollaterallyAffectedRecords(monitorPlanId: string, reportingPeriodId: number, trx?: EntityManager): Promise<void> {
    //1. Update affected EM Records
    const emResult = await withTransaction(this.repository, trx).query(
      'SELECT * FROM camdecmpswks.update_collateral_em_data_for_em_changes($1, $2)',
      [monitorPlanId, reportingPeriodId],
    );

    if (emResult[0].result === 'F') {
      throw new Error(`EM Deletion Failed: ${emResult[0].error_msg}`);
    }
  }
```

Reply:
Noted with thanks, MaxD.